sim/warrior/arms/blood_frenzy.go

```diff
@@ -25,31 +24,32 @@ func (war *ArmsWarrior) applyBloodFrenzy() {
 	bfRageProc := core.ActionID{SpellID: 92576}
 	procChance := 0.05 * float64(war.Talents.BloodFrenzy)
 	bfRageMetrics := war.NewRageMetrics(bfRageProc)
-	core.MakePermanent(war.RegisterAura(core.Aura{
-		Label: "Blood Frenzy Debuff Trigger",
-		OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
-			if !result.Landed() {
-				return
-			}
-
-			if spell.ActionID.IsOtherAction(proto.OtherAction_OtherActionAttack) {
-				if sim.Proc(procChance, "Blood Frenzy Rage Proc") {
-					war.AddRage(sim, 20, bfRageMetrics)
-				}
-			}
-
-			if spell.Flags.Matches(warrior.SpellFlagBleed) {
-				dot := spell.Dot(result.Target)
-
-				// Apply Blood Frenzy, it lasts as long as the dot is on the target
-				bf := bfAuras.Get(result.Target)
-				bf.Duration = dot.TickLength * time.Duration(dot.NumberOfTicks)
-				bf.Activate(sim)
-
-				// Apply Trauma, has fixed duration regardless of bleeds
-				trauma := traumaAuras.Get(result.Target)
-				trauma.Activate(sim)
-			}
+	core.MakeProcTriggerAura(&war.Unit, core.ProcTrigger{
+		Name:       "Blood Frenzy Rage Trigger",
+		Callback:   core.CallbackOnSpellHitDealt,
+		Outcome:    core.OutcomeLanded,
+		ProcMask:   core.ProcMaskMeleeMHAuto,
```

Comment:
Only MH attacks should trigger this? Or is this just because it being an arms talent? :D There's no way some one dual wields arms, right? :D